tests/mir-opt/box_conditional_drop_allocator.main.ElaborateDrops.diff

```diff
@@ -0,0 +1,174 @@
+- // MIR for `main` before ElaborateDrops
++ // MIR for `main` after ElaborateDrops
+  
+  fn main() -> () {
+      let mut _0: ();
+      let _1: std::boxed::Box<HasDrop, DropAllocator>;
+      let mut _2: HasDrop;
+      let mut _3: DropAllocator;
+      let mut _4: bool;
+      let _5: ();
+      let mut _6: HasDrop;
+      let _7: ();
+      let mut _8: std::boxed::Box<HasDrop, DropAllocator>;
++     let mut _9: bool;
++     let mut _10: &mut std::boxed::Box<HasDrop, DropAllocator>;
++     let mut _11: ();
++     let mut _12: &mut std::boxed::Box<HasDrop, DropAllocator>;
++     let mut _13: ();
++     let mut _14: &mut std::boxed::Box<HasDrop, DropAllocator>;
++     let mut _15: ();
+      scope 1 {
+          debug b => _1;
+      }
+  
+      bb0: {
++         _9 = const false;
+          StorageLive(_1);
+          StorageLive(_2);
+          _2 = HasDrop;
+          StorageLive(_3);
+          _3 = DropAllocator;
+          _1 = Box::<HasDrop, DropAllocator>::new_in(move _2, move _3) -> [return: bb1, unwind: bb11];
+      }
+  
+      bb1: {
++         _9 = const true;
```

Comment:
Is the placement of setting `DropAllocator`'s drop flag correct here? I would have thought it should be set immediately after `_3 = DropAllocator;` since if `Box::<HasDrop, DropAllocator>::new_in()` panics, then we will not call drop on `_3`. If you trace `bb11 -> bb12 -> bb13`, you'll see that we effectively just call `resume` in that case and `_3` isn't dropped even though it should be.

Reply:
`_3` gets moved into `new_in`. If it panics, it'll get dropped in there. The drop flag is for the allocator field, which isn't initialized until after `new_in` returns without panicking.